src/EditorFeatures/Core/Shared/Tagging/EventSources/TaggerEventSources.DiagnosticsChangedEventSource.cs

```diff
@@ -15,11 +17,13 @@ private class DiagnosticsChangedEventSource(ITextBuffer subjectBuffer, IDiagnost
             private readonly ITextBuffer _subjectBuffer = subjectBuffer;
             private readonly IDiagnosticService _service = service;
 
-            private void OnDiagnosticsUpdated(object? sender, DiagnosticsUpdatedArgs e)
+            private void OnDiagnosticsUpdated(object? sender, ImmutableArray<DiagnosticsUpdatedArgs> e)
             {
-                var documentId = e.Workspace.GetDocumentIdInCurrentContext(_subjectBuffer.AsTextContainer());
+                if (e.FirstOrDefault() is not { } first)
+                    return;
 
-                if (documentId == e.DocumentId)
+                var documentId = first.Workspace.GetDocumentIdInCurrentContext(_subjectBuffer.AsTextContainer());
```

Comment:
>Workspace [](http://example.com/codeflow?start=39&length=9)

It must be guaranteed that all items in the args will come from the same workspace?

Reply:
Right now it's written that way, but if there is a concern I could update it to no longer assume this.

Reply:
I don't have context, maybe @CyrusNajmabadi can chime on whether the assumption is safe moving forward

Reply:
yeah.  i don't know if that's safe.  it *seems* unsafe on its face, so i would write to be resilient.

Reply:
⏱️ Updating this one

Reply:
➡️ Now updated